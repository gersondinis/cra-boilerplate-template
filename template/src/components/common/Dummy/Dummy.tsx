import {FC, ReactNode} from 'react';

export const Dummy: FC<IDummyProps> = ({children}: IDummyProps) => {
  return (
    <>
      {children}
    </>
  );
};

export interface IDummyProps {
  children?: ReactNode;
}
