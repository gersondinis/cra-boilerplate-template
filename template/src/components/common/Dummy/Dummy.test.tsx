import {render} from '@testing-library/react';
import {Dummy} from 'components/common/Dummy/Dummy';


describe('Dummy', () => {
  it('renders children as expected', () => {
    expect(render(<Dummy/>).container).toMatchSnapshot();
    expect(render(<Dummy>{'children'}</Dummy>).container).toMatchSnapshot();
  });

  it('renders the provided children', () => {
    expect(render(<Dummy><span className='test'/></Dummy>).container.getElementsByClassName('test').length).toBe(1);
    expect(render(<Dummy/>).container.getElementsByClassName('test').length).toBe(0);
  });
});
