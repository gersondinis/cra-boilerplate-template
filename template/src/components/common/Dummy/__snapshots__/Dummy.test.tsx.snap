// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy renders children as expected 1`] = `<div />`;

exports[`Dummy renders children as expected 2`] = `
<div>
  children
</div>
`;
